src/sidebar/groups.js

```diff
@@ -15,16 +15,78 @@ var STORAGE_KEY = 'hypothesis.groups.focus';
 
 var events = require('./events');
 
+/**
+ * Return a value from app state when it meets certain criteria.
+ *
+ * `await` returns a Promise which resolves when a selector function,
+ * which reads values from a Redux store, returns non-null.
+ *
+ * @param {Object} store - Redux store
+ * @param {Function<T|null>} selector - Function which returns a value from the
+ *   store if the criteria is met or `null` otherwise.
+ * @return {Promise<T>}
+ */
+function awaitSelector(store, selector) {
```

Comment:
Just noting that we had a chat in Slack yesterday and agreed to change this to `awaitStateChange`.